blas/blas32/conv_symmetric_test.go

```diff
@@ -0,0 +1,307 @@
+// Code generated by "go generate gonum.org/v1/gonum/blas”; DO NOT EDIT.
+
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package blas32
+
+import (
+	math "gonum.org/v1/gonum/internal/math32"
+	"testing"
+
+	"gonum.org/v1/gonum/blas"
+)
+
+func newSymmetricFrom(a SymmetricCols) Symmetric {
+	t := Symmetric{
+		N:      a.N,
+		Stride: a.N,
+		Data:   make([]float32, a.N*a.N),
+		Uplo:   a.Uplo,
+	}
+	t.From(a)
+	return t
+}
+
+func (m Symmetric) n() int { return m.N }
+func (m Symmetric) at(i, j int) float32 {
+	if m.Uplo == blas.Lower && i < j && j < m.N {
```

Comment:
Symmetric upper truncate the leading column for col major and the trailing row for row major, and the opposite for symmetric lower.